connector/src/main/java/org/geysermc/connector/network/translators/inventory/PlayerInventoryTranslator.java

```diff
@@ -164,6 +168,11 @@ public void translateActions(GeyserSession session, Inventory inventory, List<In
             //crafting grid is not visible in creative mode in java edition
             for (InventoryActionData action : actions) {
                 if (action.getSource().getContainerId() == ContainerId.CURSOR && (action.getSlot() >= 28 && 31 >= action.getSlot())) {
+                    if (!HAS_RECEIVED_MESSAGE.contains(session.getPlayerEntity().getEntityId())) {
+                        // TODO: Allow the crafting table to be used with non-standalone versions
+                        session.sendMessage("The creative crafting table cannot be used in Java Edition.");
```

Comment:
Might want to include a bit more info here, as to players who see this may think that there is a bug present making the server think they're on Java Edition. I know this isn't the case, but to players this may come off as a "bug". Perhaps `The creative crafting table cannot be used as it's incompatible with Minecraft: Java Edition.`